### ISSUE TITLE
SaveAs overwrites existing files without question

### ISSUE THREAD

**Issue by rhiaro** [dokieli/dokieli#82]
Though this probably will be solved via #60 and #50

**Comment by csarven** [comment]
Good idea. I think #50 will help. #60 might, but not for the case where the input is done via URL entry. To prevent unwanted overwrites from actually happening, `HEAD` should be made to the URL, if 404, proceed to save to that URL, otherwise, give a warning about the resource's existence and ask the user if they want to overwrite or cancel/enter a different URL.

**Comment by csarven** [comment]
Alternatively, perhaps use `If-None-Match: *` header.

**Comment by csarven** [comment]
Hello 2016. In UIs, such as, Save As, Feeds, Reply, including components such as Create container, the final input value for the target location may already exist.

In the context where the resource browser is already loaded for a particular location (i.e., parent container/collection of the target resource location), it can already easily check if a resource with the same name already exists (without doing any additional fetches), and so the UI can be updated quickly with an indicator that the user may overwrite / replace the existing resource, e.g., change input border colour and change submit label text to "Replace".

In the case of existing containers, it shouldn't be possible to re-use the same name. So, no replace. It is not clear from the Protocol or particularly reliable at this time for dokieli to perform this. Well, technically possible, but the flow is generally not something the user my be familiar with, e.g., from their desktop or other apps.

So, if a user specifies a container name to create with an existing name, warn them, but disable the Submit/Save button.

For now, only for non-container resources it should be possible to Replace. Later we can look into introducing Delete.

In situations where we don't have the container/collection listing handy, dokieli can optimistically try to write to it with a precondition (`If-None-Match: *`). And if it exists, tell the user the filename already exists - same UI as above, input field + submit being "Replace".

When final target location doesn't conflict, change "Replace" to "Save/Submit" (or whatever is already in that particular panel/context).

FWIW, Ubuntu shows something like this:

<img width="941" height="68" alt="Screenshot of Ubuntu where filename already exists and indicator to edit the input field, with Replace button showing" src="https://github.com/user-attachments/assets/447652c6-dee2-4793-94a6-45aa228258f7" />

or

<img width="941" height="68" alt="Screenshot of Ubuntu where filename already exists and editable input field, with Replace button showing" src="https://github.com/user-attachments/assets/d2e66029-140d-4b79-b3da-3948cbbe54d7" />
